src/main/java/thePackmaster/actions/graveyardpack/PlayRandomFromExhaustAction.java

```diff
@@ -0,0 +1,122 @@
+package thePackmaster.actions.graveyardpack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInDiscardAction;
+import com.megacrit.cardcrawl.actions.utility.QueueCardAction;
+import com.megacrit.cardcrawl.actions.utility.ShowCardAndPoofAction;
+import com.megacrit.cardcrawl.actions.utility.WaitAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardQueueItem;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardBrieflyEffect;
+
+import thePackmaster.actions.graveyardpack.CleanUpCardAction;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+
+
+public class PlayRandomFromExhaustAction extends AbstractGameAction
```

Comment:
I'm sure this can be combined with the PlayFromExhaustAction

Reply:
One has to open up UI for the player and the other doesn't, so I'd rather keep them separate.

Reply:
I decided to merge them after all.